src/main/java/com/webforj/samples/views/slider/SliderTickSpacingView.java

```diff
@@ -42,44 +36,45 @@ public SliderTickSpacingView() {
         .setTooltipVisibleOnSlideOnly(true)
         .setStyle("padding", "var(--dwc-space-m) 0");
 
+    int range = slider.getMax() - slider.getMin();
+
     majorTickInput
-        .setHelperText("Enter major tick spacing (e.g., 10)")
-        .setHorizontalAlignment(NumberField.Alignment.LEFT)
-        .setMin(0d)
+        .setMin(1d)
+        .setMax((double) range)
+        .setInvalidMessage("Must be between 1 and " + range)
+        .setPlaceholder("Enter major tick spacing (e.g., 10)")
         .onValueChange(ev -> updateTickSpacing());
+
     minorTickInput
-        .setHelperText("Enter minor tick spacing (e.g., 2)")
-        .setHorizontalAlignment(NumberField.Alignment.LEFT)
-        .setMin(0d)
+        .setMin(1d)
+        .setMax((double) range)
+        .setInvalidMessage("Must be between 1 and " + range)
+        .setPlaceholder("Enter minor tick spacing (e.g., 2)")
         .onValueChange(ev -> updateTickSpacing());
 
-    snapToTicks.onToggle(ev -> {
-      isSnapToTicksEnabled = ev.isToggled();
-      slider.setSnapToTicks(isSnapToTicksEnabled);
-    });
-
-    showTicks.onToggle(ev -> {
-      areTicksVisible = ev.isToggled();
-      slider.setTicksVisible(areTicksVisible);
-    });
+    snapToTicks.onToggle(ev -> slider.setSnapToTicks(ev.isToggled()));
+    showTicks.onToggle(ev -> slider.setTicksVisible(ev.isToggled()));
 
     self.add(
         slider,
         FlexLayout.create(
             majorTickInput,
             minorTickInput,
-            FlexLayout.create(snapToTicks, showTicks).horizontal().build()).vertical().build());
+            FlexLayout.create(snapToTicks, showTicks).horizontal().build()
+        ).vertical().build()
+    );
   }
 
   private void updateTickSpacing() {
-    int majorSpacing = Optional.ofNullable(majorTickInput.getValue()).map(Double::intValue).orElse(0);
-    int minorSpacing = Optional.ofNullable(minorTickInput.getValue()).map(Double::intValue).orElse(0);
+    Double majorVal = majorTickInput.getValue();
```

Comment:
Don't reach the client for the value, pass the value from the event payload : https://docs.webforj.com/docs/architecture/client-server#improve-performance